src/condor_scripts/condor_upgrade_check

```diff
@@ -718,6 +718,42 @@ def check_desktop_policy():
     return 0
 
 
+def check_pslot_default():
+    """Check for use of RANK in startd config due to incompatibility with
+    P-Slots on by default in V23.0.0"""
+    htc_version = tuple(int(x) for x in htcondor.param["CONDOR_VERSION"].split("."))
+    if htc_version >= (23, 0, 0):
+        return 0
+    for key in htcondor.param.keys():
+        if key.lower().startswith("slot_type_"):
+            return 0
+    rank = htcondor.param.get("RANK")
+    if rank is None:
+        pass
+    else:
+        try:
+            temp = float(rank)
```

Comment:
Just noting for next time, I think you could just do this since `temp` isn't being used, it will still raise an exception.
```suggestion
            float(rank)
```